src/Provider/AbstractProvider.php

```diff
@@ -356,12 +356,7 @@ protected function fetchProviderData($url, array $headers = [])
         try {
             $client = $this->getHttpClient();
             $client->setBaseUrl($url);
```

Comment:
Looks like we can, but I'm not sure how to update the tests to reflect this change. It looks like `setBaseUrl` is being called in other places, so I'm not sure what the expected number of calls to it will be.